benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,669 @@
+#      accuracy, GenomeBiology, 16:157
+#
+
+import sys
+import click
+from os import makedirs
+from os.path import exists, basename, join, splitext
+import subprocess
+import time
+import glob
+import random
+
+from skbio import Sequence
+
+
+def extract_genbank(genbank_fp, verbose=False):
+    """Extract protein coding sequences from GenBank record.
+
+    Parameters
+    ----------
+    genbank_fp: string
+        file path to genome in GenBank format
+
+    Returns
+    -------
+    seq: skbio.sequence.Sequence
+        Sequence object
+    genes: dictionary
+        a dictionary of genes (CDS) and their info, with the key being the
+        protein IDs and the value being a 4-element list including the
+        translated sequence, the start and end positions in the genome
+    """
+    genes = {}
+    if verbose:
+        sys.stdout.write("\tParse GenBank record ...\n")
+    seq = Sequence.read(genbank_fp, format='genbank')
+    if verbose:
+        sys.stdout.write("\t\tDone.\n")
+    for feature in seq.interval_metadata.features:
+        if feature['type_'] == 'CDS':
+            protein_id = feature['protein_id']
+            translation = feature['translation']
+            strand = '-' if feature['rc_'] else '+'
+            loc = seq.interval_metadata.features[feature]
+            start_pos = loc[0][0]
+            end_pos = loc[0][1]
+            if protein_id not in genes:
+                genes[protein_id.replace("\"", "")] = [
+                    translation.replace(" ", "").replace("\"", ""),
+                    start_pos, end_pos, strand]
+            else:
+                raise KeyError("%s already exists in dictionary" % protein_id)
+    return seq, genes
+
+
+def launch_orthofinder(proteomes_dir, threads, verbose=False):
+    """Launch OrthoFinder to report orthologous gene groups for two genomes.
+
+    Parameters
+    ----------
+    proteomes_dir: string
+        directory path storing FASTA coding sequences for complete genomes
+    threads: integer
+        number of threads to use
+    verbose: boolean
+        if True, run in verbose mode
+    """
+    if verbose:
+        sys.stdout.write("\tLaunch OrthoFinder ..\n")
+    orthofinder_command = ["orthofinder.py",
+                           "-f", proteomes_dir,
+                           "-t", str(threads)]
+    proc = subprocess.Popen(orthofinder_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if stderr:
+        print(stderr)
+    if verbose:
+        print(stdout)
+        sys.stdout.write("\tDone\n")
+
+
+def _parse_orthofinder_ids(ids_fp):
+    """
+    """
+    ids = {}
+    with open(ids_fp, 'r') as ids_f:
+        for line in ids_f:
+            line = line.strip().split()
+            ids[line[0].split(':')[0]] = line[1]
```

Comment:
that would be the case, however ids in the first column are unique (and if they weren't, the ids in the second column would have to be identical for the two non-unique ids in column 1). I added though a check for unique IDs.